navigator_helpers/tasks/text_to_code/task_suite.py

```diff
@@ -59,6 +63,23 @@ def validate_code(self, code_string: str) -> str:
         message = "passed" if message is None else message
         return message
 
+    def check_semantic_correctness(self, code_string: str) -> float:
+        """Evaluate the code using pylint and return the score."""
+        pylint_output = StringIO()
+        with tempfile.NamedTemporaryFile(suffix=".py", mode="w", delete=False) as f:
```

Comment:
It may be a good idea to remove these temporary files after this check?
It looks like in this case it may be possible with `delete=True, delete_on_close=False` as per [docs](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile):

> Therefore to use the name of the temporary file to reopen the file after closing it, either make sure not to delete the file upon closure (set the delete parameter to be false) or, in case the temporary file is created in a with statement, set the delete_on_close parameter to be false. **The latter approach is recommended as it provides assistance in automatic cleaning of the temporary file upon the context manager exit.**

Reply:
Alternatively, we could have `os.unlink(..)` somewhere (e.g. `finally` block).

Reply:
Thanks. Will go with your first suggestion